### PR TITLE
Singleton Pattern

### PR DESCRIPTION
**Pull request by kondanta**
#27 *cough*
- Thread safe singleton implementation.

### CODE REVIEW

File: Taylan/despat/singleton/lab/main.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  Singleton2(const Singleton2 &) = delete;
+  Singleton2 &operator=(const Singleton2 &) = delete;
+  explicit Singleton2(string name) : AbstractClass(move(name)){};
+};
+
+class Singleton3 : public AbstractClass {
+public:
+  void printName() override { cout << "Singleton3 class"; };
+  static Singleton3 &getSingleton3() {
+    static Singleton3 instance("Singleton1");
```
Comment (sasalx) [inline Taylan/despat/singleton/lab/main.cpp]: Such heresy.

Comment (kondanta) [reply Taylan/despat/singleton/lab/main.cpp]: I cannot see any heresy over there though. :rofl:

Comment (sasalx) [reply Taylan/despat/singleton/lab/main.cpp]: c++11

Comment (kondanta) [reply Taylan/despat/singleton/lab/main.cpp]: Actually its c++17
LOL

Comment (sasalx) [reply Taylan/despat/singleton/lab/main.cpp]: I was talking about syntax.

**Pull request merged**